code/object/objcollide.cpp

```diff
@@ -1042,3 +1042,29 @@ void obj_sort_and_collide(SCP_vector<int>* Collision_list)
 	}
 	obj_find_overlap_colliders(sort_list_y, sort_list_z, 2, true);
 }
+
+void collide_apply_gravity_flags_weapons() {
+	for (int i = 0; i < MAX_OBJECTS; i++) {
```

Comment:
Why not iterate through `obj_used_list` or `Missile_obj_list`?

I suppose if primaries can home now, that rules out the second; but the first should be an option.

Reply:
Ah, true, I would've wanted a `weapon_used_list`, but `obj_used_list` is better for what I ended up going with.